test/fcntl/test_fcntl_open.c

```diff
@@ -151,8 +151,18 @@ void test() {
   errno = 0;
```

Comment:
It it worth trying to split up this main function while you are here?   It looks like at least the last 5 lines could b split into `test_creat`?

Reply:
It seems like this would take a fair amount of effort since almost all of the function is one big loop. Probably the loop body should be hoisted into a function.